### ISSUE TITLE
[Rule change] no-magic-number - allow final declaration

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#326]
**What rule do you want to change?**
no-magic-number 

**Why?**
Magic numbers often used in flutter widgets where variables can't be declared as non-static `const`.
https://dart.dev/tools/diagnostic-messages#const_instance_field

**How will the change be implemented? (New option, new default behavior, etc.)?**
Warns against using number literals outside of named constants and **finals**

**Please provide some example code that this change will affect:**

```dart
class TestCardWidget extends StatelessWidget {
  const TestCardWidget({
    Key? key,
  }) : super(key: key);

  final cardWidth = 20;
  final cardHeight = 10;

  @override
  Widget build(BuildContext _) {
    return Container(
      width: cardWidth,
      height: cardHeight,
    );
  }
}
```

**What does the rule currently do for this code?**
Warns against using number literals outside of named constants

**What will the rule do after it's changed?**
Warns against using number literals outside of named constants and **finals**

**Comment by incendial** [comment]
3.3.0 version with the improvement is now available on pub.
